app/models/archived_patient.rb

```diff
@@ -7,6 +7,8 @@ class ArchivedPatient
   # Concerns
   include Exportable
 
+  before_save :check_if_audited
```

Comment:
I'm a little worried having this hook in as a before action when it's being run in an automated fashion - if the Archive patient create fails will the patient still delete itself, or will the processing cut out?

Reply:
I think I'd prefer to take the awesome new field, and rather than invoke the process in archive save, invoke it as a third run.

Right now we run:
 - Archive Dropoff Patients
 - Archive Fulfilled Patients

But we could turn that into:

 - Archive Dropoff Patients
 - Archive _Audited_ Fulfilled Patients
 - _Archive TimedOut Fulfilled Patients_

This would involve adjusting the function [Patient:archive_fulfilled_patients](https://github.com/DCAFEngineering/dcaf_case_management/pull/1509/files#diff-d213d69335ff3bc33b8ea8545f4c4459R84) to use our time out date, and then create another similar one that grabs 120 day old fulfilled _and audited_ patients to process.

Reply:
@colinxfleming lemme know here - am I being too paranoid about the save hook?

Reply:
@lomky I think I would favor your approach of doing them in isolated batches, yeah -- adds a layer of safety and is a little more explicit about this action. Can you work with @rudietuesdays to implement that?